### PR TITLE
stress_kernel_compile: fix wget issues

### PR DESCRIPTION
**Pull request by yiqianwei**
1.Install wget tool in guest
2.Set the progress parameter for wget
3.Set wget the download time to 40 mins

ID: 2501

### CODE REVIEW

**Comment by yiqianwei** [comment]
Test results: 
 (1/1) Host_RHEL.m10.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.10.0.x86_64.io-github-autotest-qemu.stress_kernel_compile.multi_guests.q35: PASS (494.94 s)

**Comment by yiqianwei** [comment]
@yanan-fu  and @jingzhao84 , Could you help to review this patch? thanks

**Comment by yiqianwei** [comment]
Reasons to remove wget：
1. Rhel10 guest does not have wget by default
2. Even if you install wget in the guest, there is an error message in auto: Cannot create progress bar thread. Disabling progress bar

**Comment by yanan-fu** [comment]
> Reasons to remove wget：
> 
>     1.  guest does not have wget by default
> 
>     2. Even if you install wget in the guest, there is an error message in auto: Cannot create progress bar thread. Disabling progress bar

The progress bar is does not affect the download, it is not the root cause for your issue IMO.
The `wget` support  two type for progress indicator,  `none` and `bar`.  And `none` is the default behaviour when use TTY:
```
   --progress=type
       Select the type of the progress indicator you wish to use.  Supported indicator types are none and bar.

       Type bar draws an ASCII progress bar graphics (a.k.a “thermometer” display) indicating the status of retrieval.

       If the output is a TTY, bar is the default.  Else, the progress bar will be switched off, except when using --force-progress.

       The type `dot' is currently not supported, but won’t trigger an error to not break wget command lines.

       The parameterized types bar:force and bar:force:noscroll will add the effect of --force-progress.  These are accepted for better wget compatibi
```

In my test env,  download the target cause more than 18 mins while in automation the timeout value is 240s
```
time wget https://www.kernel.org/pub/linux/kernel/v2.6/linux-2.6.32.tar.bz2
linux-2.6.32.tar.bz2 100% [===========================================================================================================================================>]   61.43M   57.82KB/s
                          [Files: 1  Bytes: 61.43M [58.00KB/s] Redirects: 1  Todo: 0  Errors: 0                                                                        ]

real	18m4.961s
user	0m0.617s
sys	0m0.822s
```

I do not against use `curl`,  just please add the root cause in the commit msg, thanks!

**Comment by yiqianwei** [comment]
@yanan-fu , Even if the '-- progress=bar' parameter of wget is set and the timeout value is increased by 30 minutes, there is still hit issues:
 (1/1) Host_RHEL.m10.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.10.0.x86_64.io-github-autotest-qemu.stress_kernel_compile.multi_guests.q35: ERROR: Timeout expired while waiting for shell command to complete: 'wget https://www.kernel.org/pub/linux/kernel/v2.6/linux-2.6.32.tar.bz2 --progress=bar'    (output: "\x1b[31mCannot create progress bar thread. Disabling progress bar.\n\x1b[m[0] Downloading 'ht... (2087.79 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2024-07-29T21.59-4c1b309/results.html
JOB TIME   : 2090.45 s

**Comment by yiqianwei** [comment]
Test results:
(1/1) Host_RHEL.mxx.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.xx.0.x86_64.io-github-autotest-qemu.stress_kernel_compile.multi_guests.q35: PASS (413.35 s)

File: qemu/tests/stress_kernel_compile.py [hunk 1]
```diff
@@ -20,10 +21,13 @@ def kernelcompile(session, vm_name):
         ip = vm.get_address()
         path = params.get("download_url")
         test.log.info("kernel path = %s", path)
-        get_kernel_cmd = "wget %s" % path
+        get_kernel_cmd = "wget %s --progress=none" % path
+        if session.cmd_status("command -v wget") != 0:
+            inst_wget = utils_package.package_install("wget", session,
```
Comment (yanan-fu) [inline qemu/tests/stress_kernel_compile.py:26]: 1. It is not must to check the `wget` status,  installation will skip it automatically
2. Need check the status of `utils_package.package_install`, test.error when install failed

Comment (yiqianwei) [reply qemu/tests/stress_kernel_compile.py]: @yanan-fu , I have upated it ,  please check it. thanks

**Comment by yiqianwei** [comment]
> Call for test result with the latest code, and please do rebase to the latest master branch, thanks!

@yanan-fu , ok, I update it.

Test results:
 (1/1) Host_RHEL.mxx.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.xx.0.x86_64.io-github-autotest-qemu.stress_kernel_compile.multi_guests.q35: PASS (433.95 s)

**Review by yanan-fu — APPROVED** [review]

**Pull request merged**